dbms/src/Interpreters/executeQuery.cpp

```diff
@@ -64,6 +69,26 @@ static String joinLines(const String & query)
 }
 
 
+static String prepareQueryForLogging(const String & query, Context & context)
+{
+    String res = query;
+    // cropping is cheap and can reduce the amount of work for wipeSensitiveData
+    // side effect: if the query had long string with sensitive data which were removed,
+    // we will store less than log_queries_cut_to_length in logs
+    res = res.substr(0, context.getSettingsRef().log_queries_cut_to_length);
```

Comment:
I will change the order cropping <=> remove secrets (as cropping can remove, let's say, only last char of secret which can be enough for regexp to stop matching).